tests/src/test/scala/whisk/core/database/test/ReplicatorTests.scala

```diff
@@ -209,20 +208,23 @@ class ReplicatorTests extends FlatSpec
 
         // Trigger replication and verify the created databases have the correct format
         val (createdBackupDbs, _, _) = runReplicator(dbUrl, dbUrl, testDbPrefix, 10.minutes)
-        createdBackupDbs should have size 1
-        val backupDbName = createdBackupDbs.head
-        backupDbName should fullyMatch regex s"backup_\\d+_$dbName"
 
-        // Wait for the replication to finish
-        waitForReplication(backupDbName)
-
-        // Verify the replicated database is equal to the original database
-        compareDatabases(dbName, backupDbName, true)
-
-        // Remove all created databases
-        createdBackupDbs.foreach(removeDatabase(_))
-        createdBackupDbs.foreach(removeReplicationDoc(_))
-        removeDatabase(dbName)
+        try {
+            createdBackupDbs should have size 1
+            val backupDbName = createdBackupDbs.head
+            backupDbName should fullyMatch regex s"backup_\\d+_$dbName"
+
+            // Wait for the replication to finish
+            waitForReplication(backupDbName)
+
+            // Verify the replicated database is equal to the original database
+            compareDatabases(dbName, backupDbName, true)
+        } finally {
+            // Remove all created databases
+            createdBackupDbs.foreach(removeDatabase(_))
+            createdBackupDbs.foreach(removeReplicationDoc(_))
+            removeDatabase(dbName)
+        }
```

Comment:
Are `removeDatabase()` and `removeReplicationDoc()` resilient enough to deal with error situations that could not happen with the previous code flow?